modules/drivers/snowflake/test/metabase/driver/snowflake_test.clj

```diff
@@ -1245,3 +1246,63 @@
                (fn [^java.sql.Connection conn]
                  (is (true? (sql-jdbc.sync.interface/have-select-privilege?
                              driver/*driver* conn schema table-name))))))))))))
+
+(deftest private-key-file-updated-test
+  (mt/test-driver :snowflake
+    (let [normal-details (:details (mt/db))
+          priv-key->base64 (fn [priv-key-var]
+                             (-> (tx/db-test-env-var-or-throw :snowflake priv-key-var)
+                                 format-env-key
+                                 u/string-to-bytes
+                                 mt/bytes->base64-data-uri))
+          rsa-details (merge (dissoc normal-details :password)
+                             {:user (tx/db-test-env-var-or-throw :snowflake :pk-user)
+                              :private-key-options "uploaded"
+                              :private-key-value (priv-key->base64 :pk-private-key)
+                              :use-password false})
+          new-rsa-details (assoc rsa-details :private-key-value (priv-key->base64 :pk-private-key-2))
+          set-user-public-key (fn [pub-key-var]
+                                (let [spec (sql-jdbc.conn/connection-details->spec :snowflake normal-details)]
+                                  (jdbc/execute! spec [(format "ALTER USER %s SET RSA_PUBLIC_KEY = '%s'"
```

Comment:
I'm a bit worried about this test. Altering the user credentials is kinda beyond global: it doesn't just risk all other tests in this CI run, it risks changing credentials for the shared resource across all CI runs. Getting this wrong has the risk to lock up all snowflake tests